### ISSUE TITLE
[BLUS30608] Fight Night Champion randomly hangs.

### ISSUE THREAD

**Issue by kroaton** [RPCS3/rpcs3#6976]
Following the latest few builds, Fight Night Champion seems to be fully playable (few graphical glitches here and there) but ever so often it just freezes the emulator mid-match (I've had it happen maybe 4-5 times over a few hours of playing).

I am running the 1.02 update though the same behaviour was experienced in the base game.

[log.txt](https://github.com/RPCS3/rpcs3/files/3827919/log.txt)

**Comment by RainbowCookie32** [comment]
First: `Following the latest few builds` is almost as good as saying "at some point it broke". Find the build in which it became more unstable, otherwise it's pretty much useless to track down the regression.

Second: Attach a proper log. What you attached is a copy paste from RPCS3's window, which is not really useful. Run the game, reproduce the issue, close the emulator, and grab RPCS3.log or RPCS3.log.gz from its folder, and then upload that one instead.

**Comment by kroaton** [comment]
Will do.

**Comment by kroaton** [comment]
build: (2019-11-07)	0.0.7-9085

log.gz: [https://we.tl/t-DMgbwpx68W](url)

I've tried older builds up to #8836 and it randomly hangs in all of them.

**Comment by Augusto7743** [comment]
Try not use LLVM

**Comment by matthewhassonsonos** [comment]
Which version are you using? I'm getting locked modes for everything

**Comment by mattstermh93** [comment]
Which version are you using? Modes are all locked for me.

**Comment by Megamouse** [comment]
Please retest

**Comment by Mookie19** [comment]
I can add to this. 

on my system - AMD Ryzen 3600X/RTX2070
the game works perfectly fine (other than the FMVs and whatnot). But no freezes (played the entire story in one sitting with no problems). (TS=Off, Loop Detect = Off)

however, on my roomies system which is a I7-9700/RTX2060 his game will freeze. unfortnately I can't get you his logs. I told him to check the TSX option and disabling it seemed to freeze it almost instantly.

**Comment by 2banditos** [comment]
Hi FiGHT NiGHT fans and devs :) I have the same problem. Night Champion randomly hangs! Do you need my log file?

More important to me is the fact that the DLC's don't work... you can make the update for version 1.02 (which in my opinion sucks) but you can't install the legendary heavwyeight champions pack. This really hurts :( because im an extreme fight night fan. FNC was the reason why i bought my PS3. Other fact is that you can't use the legenderay young Foreman. He was a preorder bonus.. XBox Users can get him for free now. For Playstation Users he is not available... Maybe there is a way to unlock him? I already contacted the the EA Support but without an answer... If a RPCS3 dev could contact me i would be very happy

**Comment by Megamouse** [comment]
needs to be tested again

**Comment by 2banditos** [comment]
This error always occurs in a KO sequence
F {PPU[0x1000005] Thread (/dev_hdd0) [0x004a54f0]} VM: Access violation writing location 0x0 (unmapped memory)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11122198/RPCS3.log.gz)

**Comment by 2banditos** [comment]
·F 0:12:48.307338 {PPU[0x1000005] Thread (/dev_hdd0) [0x004a54f0]} VM: Access violation writing location 0x0 (unmapped memory)
·! 0:12:48.617734 {PPU Syscall Usage Thread} PPU: PPU Syscall Usage Stats: 
	⁂ sys_ppu_thread_yield [1472619638]
	⁂ sys_mutex_unlock [227353457]
	⁂ sys_mutex_lock [227183474]
	⁂ sys_timer_usleep [109004912]
	⁂ sys_semaphore_post [3860659]
	⁂ sys_semaphore_wait [3322591]
	⁂ sys_semaphore_trywait [3087648]
	⁂ sys_time_get_timebase_frequency [1345511]
	⁂ sys_event_queue_receive [1225089]
	⁂ sys_mutex_trylock [170007]
	⁂ sys_event_queue_tryreceive [138329]
	⁂ sys_cond_signal [117688]
	⁂ sys_ppu_thread_set_priority [110519]
	⁂ sys_net_bnet_poll [91378]
	⁂ sys_rsx_attribute [87052]
	⁂ sys_rsx_context_attribute [43563]
	⁂ sys_net_bnet_accept [43493]
	⁂ sys_ppu_thread_get_priority [36839]
	⁂ sys_cond_wait [25894]
	⁂ sys_mutex_create [11103]
	⁂ sys_mutex_destroy [10580]
	⁂ sys_cond_create [10495]
	⁂ sys_cond_destroy [10451]
	⁂ sys_fs_read [10334]
	⁂ sys_semaphore_create [7690]
	⁂ sys_semaphore_destroy [7204]
	⁂ sys_fs_lseek [5175]
	⁂ sys_fs_write [2818]
	⁂ _sys_lwmutex_lock [1987]
	⁂ _sys_lwmutex_unlock [1987]
	⁂ sys_fs_fcntl [275]
	⁂ sys_event_queue_drain [195]
	⁂ sys_fs_open [157]
	⁂ sys_fs_close [108]
	⁂ sys_fs_stat [106]
	⁂ sys_fs_readdir [82]
	⁂ _sys_lwmutex_create [67]
	⁂ _sys_ppu_thread_create [58]
	⁂ sys_ppu_thread_start [58]
	⁂ sys_fs_opendir [56]
	⁂ _sys_prx_start_module [48]
	⁂ sys_fs_closedir [39]
	⁂ _sys_ppu_thread_exit [34]
	⁂ sys_mmapper_allocate_shared_memory [29]
	⁂ sys_mmapper_map_shared_memory [29]
	⁂ sys_time_get_current_time [24]
	⁂ sys_memory_allocate [24]
	⁂ sys_memory_get_page_attribute [21]
	⁂ _sys_prx_load_module [17]
	⁂ sys_memory_get_user_memory_size [16]
	⁂ _sys_lwmutex_destroy [15]
	⁂ sys_spu_thread_connect_event [12]
	⁂ sys_net_infoctl [10]
	⁂ sys_fs_mkdir [9]
	⁂ sys_spu_thread_group_create [6]
	⁂ sys_spu_thread_initialize [6]
	⁂ sys_spu_thread_group_start [6]
	⁂ sys_time_get_timezone [5]
	⁂ sys_process_getpid [4]
	⁂ sys_process_get_sdk_version [3]
	⁂ sys_ppu_thread_join [3]
	⁂ sys_cond_signal_all [3]
	⁂ _sys_prx_register_library [3]
	⁂ sys_rsx_context_iomap [3]
	⁂ sys_net_bnet_socket [3]
	⁂ sys_net_abort [3]
	⁂ sys_ppu_thread_rename [2]
	⁂ sys_event_flag_create [2]
	⁂ sys_event_queue_create [2]
	⁂ _sys_prx_register_module [2]
	⁂ sys_net_bnet_setsockopt [2]
	⁂ sys_net_bnet_close [2]
	⁂ sys_fs_rmdir [2]
	⁂ sys_ss_access_control_engine [2]
	⁂ _sys_process_get_paramsfo [1]
	⁂ sys_event_port_create [1]
	⁂ sys_event_port_connect_local [1]
	⁂ sys_memory_container_create [1]
	⁂ sys_mmapper_allocate_address [1]
	⁂ syscall_462 [1]
	⁂ _sys_prx_load_module_list [1]
	⁂ sys_rsx_memory_allocate [1]
	⁂ sys_rsx_context_allocate [1]
	⁂ sys_rsx_device_map [1]
	⁂ sys_net_bnet_bind [1]
	⁂ sys_net_bnet_connect [1]
	⁂ sys_net_bnet_listen [1]
	⁂ sys_net_bnet_shutdown [1]
·! 0:12:50.782302 {RSX [0x0017918]} RSX: Swapchain: present mode 2 in use.
·! 0:12:50.782537 {RSX [0x0017918]} RSX: Swapchain: requesting full screen exclusive mode 0.
·! 0:12:51.865515 {RSX [0x0017918]} RSX: Swapchain: present mode 2 in use.
·! 0:12:51.865745 {RSX [0x0017918]} RSX: Swapchain: requesting full screen exclusive mode 0.
·! 0:12:52.841635 {Performance Sensor} SYS: CPU Usage: Total: 27.3%, Cores: 34.4%, 20.3%, 28.2%, 42.2%, 28.2%, 9.4%, 37.5%, 6.3%, 61.0%, 21.9%, 23.5%, 31.3%, 23.5%, 28.2%, 36.0%, 4.7%
·! 0:12:58.133289 GUI: Game window close event issued
·! 0:12:58.133350 SYS: Stopping emulator...

**Comment by 2banditos** [comment]
It's a pitty that Fight Night Champion since years is almost unplayable
The game freezes in almost every second match...

**Comment by digant73** [comment]
@2banditos is that issue still present?

**Comment by digant73** [comment]
tested the demo. In my case there is no issue on current build 0.0.38-18327 but probably it is better that other users provide their results

**Comment by Satan86** [comment]
i think i made the full game crash if i customised my character in legacy mode too much, i need to try again tho its been a year or two

ill try to make it crash again and let u know

**Comment by Mookie19** [comment]
It's been a long time. But I figured I'd give an update from someone who's tested this game for years at this point.

Fight Now - works with every fighter, including DLC fighters. (Boardwalk hall venue 100% works, I can't remember if any venue has any crashes anymore, I will update in a couple days about this)

Bare knuckle Mode (DLC) - works 100% and 'the pen' venue works aswell.

Champions mode (Story Mode) - fully works, cutscenes audio, etc. I can't say if there's any problems after the last Issac Frost match because its impossible for me lol.

Legacy Mode (Create a fighter career) - half works the last time I tried it (probably about a year ago). The only problem i remember having was visual artifacts (black character models, etc) which i believe has been fixed since I no longer require 'write color buffer'. The only other thing was the starting tournament would hang/freeze... the solution was to just skip/sim that first amateur tournament. (I will also test and update in a few days). Oh and I think some of the training crashes the game so you have to sim that too iirc.

Create a fighter - no idea what works and what doesn't i don't make CAFs.

When I get a chance to test the game again I will update this with thread with my findings on the current RPCS build.

**Comment by digant73** [comment]
thanks for the reply. Waiting for your update

**Comment by Mookie19** [comment]
Alright, update time. 

running RPCS3 .38-18327 
5800X
32GB 3200mhz
2070 super (latest drivers)
RPCS3 on SSD (game is located on a HDD)

Game is using pretty much everything default settings wise apart from a resolution upscale to 1440p. both LLVM for SPU/PPU decoder, no SPU loop detection, Re-bar is enabled (that must be new). and is using patch 1.02.

I'd personally say this game is 100% playable and if not 95%. 

I didn't encounter one problem with my testing, no crashes, freezes or any PPU/SPU errors.

I tried fight now with Jack Dempsey vs Sonny Liston at the 'New York Stadium' no issues.
next fight (to test multiple fights with no restarts since someone said it hangs after a few fights) Rocky Marciano vs Lennox Lewis at the MGM stadium, again no problems.

I then tried the bareknuckle mode, again rocky and lewis, at the 'cowboy stadium', again no problems. One 'graphical' glitch did occur, where the round time and both fighters stamina bars disappeared. Unsure though if thats an RPCS3 bug or a glitch within FNCs itself, regardless It doesn't impact the game majorly (I prefer no hud anyways)

I then tried the Legacy career mode, which I stated used to have a problem with the intial amateur tournment not loading (infinite loading) or hanging/freezing on load. I'm happy to announce that this issue is no longer present. All 4 fights in the tournament loaded without problems. I then did 1 more fight to test the training games and I can confirm both the 'double bag' and 'stay on your feet' training minigames work flawlessly. I would go out on a limb and say that 'open sparring' and 'Heavy bag' minigames all work. 

All of this is also with zero audio issues and graphical artifacts.
I did not bother checking the create a fighter potential issues, I did however load up a fight now with a CAF I created years ago (so it must've worked at some point). And my CAF was fully rendered with no visual bugs. The only thing is the character portrait does not show the character during fighter selection. Not a big deal, and might be due to the CAF being made before an RPCS3 update that might've fixed custom character rendering. 

I re-read some old comments on this thread... specifically this one: 

"This error always occurs in a KO sequence
F {PPU[0x1000005] Thread (/dev_hdd0) [0x004a54f0]} VM: Access violation writing location 0x0 (unmapped memory)"

I knocked down my opponent multiple times (thus eventually leading to a KO) in all 3 fight now matches. I would say that issue is fixed, or was specific to that individual due to hardware limitations/capability issues.

TLDR: All in all, I'm quite confident (on my system) that this game is 100% working or 95% working with minimal bugs and errors that would lead to an unplayable game. 

If any more testing is wanted I can try. Also if you'd like a log (not sure what it would show anything since the game works) but I can provide one if needed.

Enjoy the fights for whoever is reading this... I'm sure a lot of Fight Night Forever people will start arriving here soon and hopefully this info helps them.

**Comment by digant73** [comment]
@Mookie19 thanks for the in deep checks!

**Comment by Satan86** [comment]
yeah i play legacy mode a lot, and i gave my character lots of customisation and the game worked fine, and fyi @Mookie19 the portraits are fixed with write colour buffers, but yeah no crashes, this issue is fixed

**Comment by digant73** [comment]
@Satan86 thanks. If WCB is required, wiki page should also be updated. I do not see WCB reported as mandatory setting

**Comment by digant73** [comment]
edited wiki page adding WCB as mandatory Provided comment `Fixes missing portraits on some menus`

**Comment by Triticum0** [comment]
Sorry to inform you but was still able to reproduce this issue.

[FIGHT NIGHT CHAMPION DEMO \[NPUB90513\].log.gz](https://github.com/user-attachments/files/24682933/FIGHT.NIGHT.CHAMPION.DEMO.NPUB90513.log.gz)

F {PPU[0x1000005] Thread (/dev_hdd0) [0x0049d6b8]} VM: Access violation writing location 0x0 (unmapped memory)

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/29a3bb15-8e16-415e-b0ff-2f0b48664134" />

Note: Could only reproduce this during first boot of game

@digant73

**Comment by 2banditos** [comment]
F {PPU[0x1000005] Thread (/dev_hdd0) [0x0049d6b8]} VM: Access violation writing location 0x0 (unmapped memory)

Unfortunately, the error still appears. The error appears when a boxer has been knocked out!

**Comment by 2banditos** [comment]
> Alright, update time.
> 
> running RPCS3 .38-18327 5800X 32GB 3200mhz 2070 super (latest drivers) RPCS3 on SSD (game is located on a HDD)
> 
> Game is using pretty much everything default settings wise apart from a resolution upscale to 1440p. both LLVM for SPU/PPU decoder, no SPU loop detection, Re-bar is enabled (that must be new). and is using patch 1.02.
> 
> I'd personally say this game is 100% playable and if not 95%.
> 
> I didn't encounter one problem with my testing, no crashes, freezes or any PPU/SPU errors.
> 
> I tried fight now with Jack Dempsey vs Sonny Liston at the 'New York Stadium' no issues. next fight (to test multiple fights with no restarts since someone said it hangs after a few fights) Rocky Marciano vs Lennox Lewis at the MGM stadium, again no problems.
> 
> I then tried the bareknuckle mode, again rocky and lewis, at the 'cowboy stadium', again no problems. One 'graphical' glitch did occur, where the round time and both fighters stamina bars disappeared. Unsure though if thats an RPCS3 bug or a glitch within FNCs itself, regardless It doesn't impact the game majorly (I prefer no hud anyways)
> 
> I then tried the Legacy career mode, which I stated used to have a problem with the intial amateur tournment not loading (infinite loading) or hanging/freezing on load. I'm happy to announce that this issue is no longer present. All 4 fights in the tournament loaded without problems. I then did 1 more fight to test the training games and I can confirm both the 'double bag' and 'stay on your feet' training minigames work flawlessly. I would go out on a limb and say that 'open sparring' and 'Heavy bag' minigames all work.
> 
> All of this is also with zero audio issues and graphical artifacts. I did not bother checking the create a fighter potential issues, I did however load up a fight now with a CAF I created years ago (so it must've worked at some point). And my CAF was fully rendered with no visual bugs. The only thing is the character portrait does not show the character during fighter selection. Not a big deal, and might be due to the CAF being made before an RPCS3 update that might've fixed custom character rendering.
> 
> I re-read some old comments on this thread... specifically this one:
> 
> "This error always occurs in a KO sequence F {PPU[0x1000005] Thread (/dev_hdd0) [0x004a54f0]} VM: Access violation writing location 0x0 (unmapped memory)"
> 
> I knocked down my opponent multiple times (thus eventually leading to a KO) in all 3 fight now matches. I would say that issue is fixed, or was specific to that individual due to hardware limitations/capability issues.
> 
> TLDR: All in all, I'm quite confident (on my system) that this game is 100% working or 95% working with minimal bugs and errors that would lead to an unplayable game.
> 
> If any more testing is wanted I can try. Also if you'd like a log (not sure what it would show anything since the game works) but I can provide one if needed.
> 
> Enjoy the fights for whoever is reading this... I'm sure a lot of Fight Night Forever people will start arriving here soon and hopefully this info helps them.

Unfortunately, the error "Access violation writing location 0x0 (unmapped memory)" still appears. The error appears when a boxer has been knocked out. This does not happen with every knockout, but the error appears sooner or later in "fight now" mode... That's really annoying.

**Comment by 2banditos** [comment]
Following Augusto7743's advice, I switched the SPU decoder from LLVM to ASMJIT. I have fought 22 boxing matches in a row! Without a single freeze. It does indeed appear that the error only affects LLVM mode. I can't confirm it 100 percent but it could be a small hint

**Comment by digant73** [comment]
please an update on current build

**Comment by keikof070623-gif** [comment]
Can anyone test if this is still an issue and if switching from LLVM to another decoder fixes this? Perhaps this is another game that does not work with LLVM and it can be added to that list
